src/common/pagination.py

```diff
@@ -12,102 +12,162 @@
 # language governing permissions and limitations under the License.
 
 from dataclasses import dataclass
-from datetime import timedelta
+from datetime import timedelta, datetime, timezone
 import base64
 import json
-from typing import Union, Optional
-
-import boto3
-import aws_encryption_sdk
+import os
+from typing import Optional, ClassVar, Any, Tuple
 
 import aws_lambda_api_event_utils as api_utils
 
+from .event_utils import get_caller_identity, get_api_id, get_path
+from .encryption import EncryptionClient
+
 
 class PaginationTokenError(api_utils.APIErrorResponse):
     STATUS_CODE = 400
     ERROR_CODE = "InvalidNextToken"
     ERROR_MESSAGE = "The provided NextToken is invalid."
 
 
-@dataclass
-class EncryptionClient:
-    client: aws_encryption_sdk.EncryptionSDKClient
-    key_provider: aws_encryption_sdk.key_providers.base.MasterKeyProvider
-    cache: aws_encryption_sdk.caches.base.CryptoMaterialsCache
-    materials_manager: aws_encryption_sdk.materials_managers.base.CryptoMaterialsManager
-
-    def encrypt(self, plaintext: Union[str, bytes]) -> bytes:
-        ciphertext, header = self.client.encrypt(
-            source=plaintext,
-            materials_manager=self.materials_manager,
-        )
-        return ciphertext
-
-    def decrypt(self, ciphertext: Union[str, bytes]) -> bytes:
-        plaintext, header = self.client.decrypt(
-            source=ciphertext,
-            materials_manager=self.materials_manager,
+@dataclass(frozen=True)
+class PaginationTokenContext:
+    """Represents a context to include in the pagination token.
+    The context in a token can then be validated against the current context,
+    to require that token cannot be reused across contexts."""
+    context: dict
+    expiration: datetime
+    DISABLE_PAGINATION_CONTEXT_STRICT_VALIATION_ENV_KEY: ClassVar[
```

Comment:
nit: typo (throughout)

```suggestion
    DISABLE_PAGINATION_CONTEXT_STRICT_VALIDATION_ENV_KEY: ClassVar[
```